assets/src/components/activities/adaptive/components/authoring/LayoutEditor.tsx

```diff
@@ -37,97 +43,86 @@ const defaultHandler = async () => {
 
 const toolBarTopOffset = -38;
 
+const getPartAndCapabilities = (
+  selectedPartId: string,
+  parts: AnyPartComponent[],
+): (AnyPartComponent & { capabilities: PartCapabilities }) | null => {
+  const part = parts.find((p) => p.id === selectedPartId);
+  if (part) {
+    let capabilities = { ...defaultCapabilities };
+    // attempt to get an instance of the part class
+    const PartClass = customElements.get(part.type);
+    if (PartClass) {
+      // TODO: cache the instance data somewhere so we don't do this every time
+      const instance = new PartClass() as any; // TODO: extend HTMLElement?
+      if (instance.getCapabilities) {
+        capabilities = { ...capabilities, ...instance.getCapabilities() };
+      }
+      const partWithCapabilities = { ...part, capabilities };
+      return partWithCapabilities;
+    }
+  }
+  return null;
+};
+
 const LayoutEditor: React.FC<LayoutEditorProps> = (props) => {
   const pusherContext = useContext(NotificationContext);
-  const [pusher, setPusher] = useState(pusherContext || new EventEmitter().setMaxListeners(50));
 
-  useEffect(() => {
-    if (pusherContext) {
-      setPusher(pusherContext);
-    }
-  }, [pusherContext]);
+  const pusher = useMemo(
+    () => pusherContext || new EventEmitter().setMaxListeners(50),
+    [pusherContext],
+  );
+
+  // The size of the current component *while* it's being resized before new size is, ignored if not actively resizing
+  const [dragSize, setDragSize] = useState({ width: 0, height: 0 });
+  const [isDragging, setIsDragging] = useState(false);
 
   const [parts, setParts] = useState(props.parts);
-  const [selectedPartId, setSelectedPartId] = useState(props.selected || '');
+
   const [configurePartId, setConfigurePartId] = useState('');
-  const [selectedPart, setSelectedPart] = useState<any>(null);
-  const [isDragging, setIsDragging] = useState(false);
+
+  const selectedPartId = props.selected || '';
```

Comment:
Some more useState refactors, the currently selected part is really controlled higher up in the component chain, and this value would sometimes be out of sync here. 

I saw at least one bug with a component that was on a containing layer could stay selected when going into a screen. This fixes that sort of bug.